rust/defi/src/defi_dapp/exchange.rs

```diff
@@ -136,33 +107,45 @@ impl Exchange {
         to_amount: Nat,
     ) -> OrderPlacementReceipt {
         ic_cdk::println!("place order");
+        if from_amount <= 0u8 || to_amount <= 0u8 {
```

Comment:
Why 0u8 here and 0u16 elsewhere?

Reply:
Just a lack of consistency, this has all been replaced with explicit construction.